float64 rotate_angle         # The only difference between the actions is the rotation angle around z-axis.
float64 x
float64 y
float64 z               # The hight to put the block on the frame